x16-smc.ino

```diff
@@ -572,13 +641,35 @@ void mouseClockIrq() {
 // ----------------------------------------------------------------
 // Bootloader Startup
 // ----------------------------------------------------------------
-void startBootloader() {
-  if (bootloaderFlags == 3 && bootloaderTimer > 0) {
-    ((void(*)(void))BOOTLOADER_START_ADDR)();
-  }
-  else {
-    bootloaderTimer = 50;
+void evaluateBootloaderKeypress() {
+  if (bootloaderTimer > 0 && specialModeActive != 1) {
+    uint8_t temp = SREG;
+    cli();
+    bootloaderTimer--;
+    SREG = temp;
+
+    uint8_t currentKeysPressed = digitalRead_opt(POWER_BUTTON_PIN) | (digitalRead_opt(RESET_BUTTON_PIN) << 1) | (digitalRead_opt(NMI_BUTTON_PIN) << 2);
+    currentKeysPressed ^= 7; // Buttons are active low. Flip bits, to let the following logic use active high logic instead (more intuitive)
+
+    // Reduce timer to 0.5 s once the first key is pressed down
+    if (currentKeysPressed && bootloaderTimer > 50) {
+      bootloaderTimer = 50;
+    }
+
+    // specialModeRequested == 0 -> try to start bootloader (Power(0x01) + Reset(0x02) + no NMI)
+    if (specialModeRequested == 0 && currentKeysPressed == 3) { // Power (1) + Reset (2) + no NMI
+      cli();
+      ((void(*)(void))BOOTLOADER_START_ADDR)(); // Start bootloader
+    }
```

Comment:
Note that not all variants of the X16 board have an NMI button. And a lot of cases only have two buttons. As the NMI button is the least useful of the three buttons, it's like not connected. Would it be more convenient to use Power+Reset for both modes?